lib/IAMClient.js

```diff
@@ -134,22 +146,42 @@ class VaultClient {
      *
      * @param {string} accountName - account name
      * @param {VaultClient~requestCallback} callback - callback
+     * @param {object} options - additional creation params
      * @returns {undefined}
      */
-    generateAccountAccessKey(accountName, callback) {
+    generateAccountAccessKey(accountName, callback, options) {
         assert(typeof accountName === 'string' && accountName !== '',
             'accountName is required');
+        const data = {
+            Action: 'GenerateAccountAccessKey',
+            Version: '2010-05-08',
+            AccountName: accountName,
+        };
+        if (options !== undefined) {
+            if (options.externalAccessKey !== undefined
+                && options.externalAccessKey !== null) {
+                assert(typeof options.externalAccessKey === 'string'
+                    && regexpAccessKey.test(options.externalAccessKey)
+                    && options.externalAccessKey !== '',
```

Comment:
minor: I think an empty string would not be accepted by the regular expression anyway.